### PR TITLE
Add 'adapter' and 'descriptor' command to istioctl mixer.

### PR DESCRIPTION
**Pull request by jmuk**

### CODE REVIEW

File: cmd/istioctl/mixer.go [hunk 1]
```diff
@@ -155,37 +164,63 @@ istioctl mixer rule delete global myservice.ns.svc.cluster.local
...
+	}
 
-	path := mixerRulePath(scope, subject)
-	status, body, err := mixerRESTRequester.Request(http.MethodPut, path, rule)
+	mixerAdapterCreateCmd = &cobra.Command{
+		Use:   "create <scope>",
+		Short: "Create Istio Mixer adapters",
+		Example: `
+# Create new Mixer adapter configs for the given scope.
+istioctl mixer adapter create global -f mixer-rule.yml
```
Comment (mandarjog) [inline cmd/istioctl/mixer.go]: `-f adapters.yaml`

File: cmd/istioctl/mixer.go [hunk 2]
```diff
@@ -155,37 +164,63 @@ istioctl mixer rule delete global myservice.ns.svc.cluster.local
...
+		SilenceUsage: true,
+	}
 
-		fmt.Printf("%s\n", message)
+	mixerDescriptorCreateCmd = &cobra.Command{
+		Use:   "create <scope>",
+		Short: "Create Istio Mixer descriptors",
+		Example: `
+# Create new Mixer descriptor configs for the given scope.
+istioctl mixer descriptor create global -f mixer-rule.yml
```
Comment (mandarjog) [inline cmd/istioctl/mixer.go]: `-f descriptors.yml`

File: cmd/istioctl/mixer.go [hunk 3]
```diff
@@ -205,28 +240,81 @@ func mixerRuleGet(scope, subject string) (string, error) {
...
 		var response mixerAPIResponse
 		message := "unknown"
-		if errJSON := json.Unmarshal(body, &response); errJSON == nil {
+		if errJSON := json.Unmarshal(respBody, &response); errJSON == nil {
 			message = response.Status.Message
 		}
 
 		if status != http.StatusOK {
-			return fmt.Errorf("failed rule deletion with status %v: %s", status, message)
+			return fmt.Errorf("failed rule creation with status %v: %s", status, message)
```
Comment (ayj) [inline cmd/istioctl/mixer.go]: `mixerRequest` is also called by `mixerRuleDelete` in which case this error message is incorrect. Maybe just generalize this with the following?

```golang
return fmt.Errorf("request failed with status: %v: %v", status, message)
```

Comment (mandarjog) [inline cmd/istioctl/mixer.go]: This is common code now. 
the error should be `failed to {method}   {path}`

File: cmd/istioctl/mixer.go (hunk 1 above)
Comment (jmuk) [inline cmd/istioctl/mixer.go]: done

File: cmd/istioctl/mixer.go (hunk 2 above)
Comment (jmuk) [inline cmd/istioctl/mixer.go]: done

File: cmd/istioctl/mixer.go (hunk 3 above)
Comment (jmuk) [inline cmd/istioctl/mixer.go]: modified to @mandarjog 's suggestion. thanks.

File: cmd/istioctl/mixer.go [hunk 4]
```diff
@@ -205,28 +240,83 @@ func mixerRuleGet(scope, subject string) (string, error) {
...
+func mixerRuleGet(scope, subject string) (string, error) {
+	return mixerGet(mixerRulePath(scope, subject))
+}
+
+func mixerRuleDelete(scope, subject string) error {
+	return mixerRequest(http.MethodDelete, mixerRulePath(scope, subject), nil)
+}
+
+func mixerAdapterOrDescriptorPath(scope, name string) string {
+	return fmt.Sprintf("api/v1/scopes/%s/%s", scope, name)
```
Comment (mandarjog) [inline cmd/istioctl/mixer.go]: I think we should make mixerBasePath = "api/v1" a constant.

Comment (jmuk) [inline cmd/istioctl/mixer.go]: Done.

**Comment by jmuk** [comment]
Also added `url.PathEscape` for the input.

**Pull request merged**